src/Orleans/Core/GrainFactory.cs

```diff
@@ -172,13 +192,29 @@ public Task<TGrainObserverInterface> CreateObjectReference<TGrainObserverInterfa
 
             Delegate creator;
 
-            if (!referenceCreators.TryGetValue(interfaceType, out creator))
+            IGrainMethodInvoker invoker;
+            if (!this.invokers.TryGetValue(interfaceType, out invoker))
+            {
+                invoker = MakeInvoker(interfaceType);
+                if (invoker != null)
+                {
+                    this.invokers.TryAdd(interfaceType, invoker);
+                }
+            }
+
+            if (invoker != null)
```

Comment:
Is it even possible for `invoker` to be null here? Shouldn't we throw if for some reason it is? Or is there a legitimate case when it can be null?